packages/compass-assistant/src/compass-assistant-provider.tsx

```diff
@@ -48,6 +48,10 @@ export type AssistantMessage = UIMessage & {
       description: string;
       state: 'confirmed' | 'rejected' | 'pending';
     };
+    /** Overrides the default sent instructions for the assistant for this message. */
+    instructions?: string;
```

Comment:
I still think that overriding instructions will have to make it confused and hallucinate because the whole conversation (ie. all the context) will make no sense anymore.